project-koku/koku
# Push cost management application status to platform on create/edit

Platform Sources recently enhanced their API to allow for applications to set source status (https://github.com/ManageIQ/sources-api/pull/116)

This change make the Sources Client push the source status to the platform on source create and edit.  The error messages being saved on the platform is currently the detailed string from the `/provider` 400 error response.  I will be opening a docs/UX issue to review these messages to ensure they are consistent with the platform and we are not inadvertently revealing any sensitive information.

**Testing**
1. Add an AWS Source with invalid bucket name and veirfy that the platform sources /application response has status strings set correctly and the koku provider has not been created yet.
2.  Edit the source with the correct bucket name and verify the koku provider is created (and is active).  Verify that the avaiable status is set on platform sources
3.  Edit the AWS source with the wrong bucket.  Verify status is updated in platform sources and that the koku provider is marked not active.
4.  Edit bucket again with correct bucket and verify provider is active and platform sources status is avaiable.

**Test Results**
[sources_push_status_ut.txt](https://github.com/project-koku/koku/files/3809378/sources_push_status_ut.txt)

Addresses #1354
